draft-ietf-jsonpath-base.md

```diff
@@ -1402,91 +1454,125 @@ According to {{filter-selector}}, a `function-expr` is valid as a `filter-path`
 or a `comparable`.
 
 Any function expressions in a query must be well-formed (by conforming to the above ABNF)
-and correctly typed,
+and well-typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
-To define which function expressions are correctly typed,
+To define which function expressions are well-typed,
 a type system is first introduced.
 
 ### Type System for Function Expressions {#typesys}
 
 Each argument and result of a function extension must have a declared type.
 
-A type is a set of instances. A type is a subtype of another type if its set of instances (possibly after coercion)
-is a subset of the set of instances of the other type.
-The subtype relationship is transitive: for any A, B, and C if B is a subtype of A and C is a subtype of B, then C is a subtype of A.
-We denote a type `T` and the transitive closure of all its subtypes as `ST(T)`.
+A type is a set of instances.
+Declared types enable testing a JSONPath query for well-typedness
+independent of any argument the JSONPath query is applied to.
 
 {{tbl-types}} defines the available types in terms of abstract instances, where `n` denotes a node, `v` denotes a value, and `nl` denotes
-a non-empty nodelist. The table also lists the (immediate) subtypes of each type.
-
-| Type                      | Abstract Instances                       | Subtypes                                       |
-| :--                       | :----------------                        | :------                                        |
-| `OptionalNodeOrValueType` | `Node(n)`, `Value(v)`, `Nothing`         | `OptionalNodeType`, `OptionalValueType`        |
-| `OptionalNodeType`        | `Node(n)`, `Nothing`                     |                                                |
-| `OptionalValueType`       | `Value(v)`, `Nothing`                    | `OptionalNodeType`, `ValueType`, `OptionalBooleanType` |
-| `ValueType`               | `Value(v)`                               | `BooleanType`                                  |
-| `OptionalBooleanType`     | `Value(true)`, `Value(false)`, `Nothing` | `BooleanType`                                  |
-| `BooleanType`             | `Value(true)`, `Value(false)`            |                                                |
-| `OptionalNodesType`       | `Nodes(nl)`, `Nothing`                   | `OptionalNodeType`                             |
+a non-empty nodelist.
+
+| Type                 | Abstract Instances                       |
+| :--                  | :----------------                        |
+| `ValueType`          | `Value(v)`, `Nothing`                    |
```

Comment:
I prefer `OptionalValueType` as this gives a hint of the possibility of the value `Nothing`.

Reply:
I tend to not get hung up on naming, and while I don't fully disagree with you, I think `ValueType` is considerably less clunky than `OptionalValueType`.  `ValueType` being defined here in this way is sufficient for me.